src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php

```diff
@@ -46,6 +50,26 @@ public function process(ContainerBuilder $container)
             }
             $definition->innerServiceId = $renamedId;
 
+            if (!$container->has($inner)) {
+                $invalidBehavior = $decorated[3] ?? ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE;
+
+                if (ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE === $invalidBehavior) {
```

Comment:
this should be the "default" case, when $invalidBehavior is neither "ignore" nor "null" - defensive programming rulez.